tests/python/pytest_input_generators.py

```diff
@@ -1524,18 +1524,21 @@ def linear_input_generator(
         requires_grad=requires_grad,
     )
 
-    def multiply_range(maximum, step):
-        assert maximum % step == 0
-        num_steps = int(math.log(maximum, step))
-        return tuple(
-            map(pow, itertools.repeat(step, num_steps), range(1, num_steps + 1))
-        )
+    B = 64
+    M = 512
+    N = 256
+    K = 32
+
+    # Cases without bias
+    shapes_input = ((K), (M, K), (B, M, K), (B, 1, M, K))
+    shapes_weight = ((K), (N, K), (1, K))
+    for shape_input, shape_weight in itertools.product(shapes_input, shapes_weight):
+        yield SampleInput(make_arg(shape_input), make_arg(shape_weight))
 
-    # Ranges of tensor sizes: 8, 64, 512, 4096, 32768, ...
-    # Use a Cartesian product to create a wide range of matrix shapes
-    # I'll stop at 512 as possible numerical difference may show up.
-    M, N, K = itertools.repeat(multiply_range(512, 8), 3)
-    for M, N, K in itertools.product(M, N, K):
-        lhs_shape = (M, K)
-        rhs_shape = (N, K)
-        yield (SampleInput(make_arg(lhs_shape), make_arg(rhs_shape), make_arg((N,))))
+    # Cases with bias
+    shape_weight = (N, K)
+    shapes_bias = ((), (N,))
+    for shape_input, shape_bias in itertools.product(shapes_input, shapes_bias):
+        yield SampleInput(
+            make_arg(shape_input), make_arg(shape_weight), make_arg(shape_bias)
+        )
```

Comment:
can we do an error_input_generator for linear as well so we have legit error message thrown. Sorry I keep nitpicking on these corner cases.

Reply:
I don't have any other concern out side of test coverage for the `unsupported` cases.

I think we want to see exceptions when
`1d weight + bias`
`2d weight (size-1 out feature) + bias (not size-1 out feature)`

Reply:
IIUC, error_generator will check if we the error message matches.

For the second case though, I am relying on torch to throw the error since we may have symbolic tensors and cannot assert if the extents will be the same.

Reply:
Oh nvm, I can match the torch errors too.